src/gui/layout/qgslayoutpicturewidget.cpp

```diff
@@ -472,7 +473,7 @@ void QgsLayoutPictureWidget::sourceChanged( const QString &source )
   if ( mPicture )
   {
     mPicture->beginCommand( tr( "Change Picture" ) );
-    mPicture->setPicturePath( source, QgsLayoutItemPicture::FormatSVG );
```

Comment:
You should be explicitly stating the type here -- why the change?

Reply:
if you are in Raster mode, the image is not displayed then.

would this be better to you:
    `mPicture->setPicturePath( source, mRadioSVG->isChecked() ? QgsLayoutItemPicture::FormatSVG : QgsLayoutItemPicture::FormatRaster );`

but this whole think svg/raster is really painful, though I don't have any simple proposition to improve the situation.

Reply:
> would this be better to you:
mPicture->setPicturePath( source, mRadioSVG->isChecked() ? QgsLayoutItemPicture::FormatSVG : QgsLayoutItemPicture::FormatRaster );

Yes please.

> but this whole think svg/raster is really painful, though I don't have any simple proposition to improve the situation.

I didn't have a good solution either, which was why I pushed it to make the caller responsibility for setting it!